homeassistant/components/zha/core/channels/smartenergy.py

```diff
@@ -210,18 +210,22 @@ def get_formatting(formatting: int) -> str:
 
         return f"{{:0{width}.{r_digits}f}}"
 
-    def _formatter_function(self, selector: FormatSelector, value: int) -> int | float:
+    def _formatter_function(
+        self, selector: FormatSelector, value: int
+    ) -> int | float | str:
```

Comment:
`self._summa_format.format(value_float).lstrip()` and `self._format_spec.format(value_float).lstrip()` will return a string.
Do we need to convert those to floats? or is it correct that they return a string?

Reply:
Good question. IIRC the only reason this is formatted as string, because per spec some formatting may have leading 0, e.g. `0001.2`, but i don't think i saw any of it in the wild. 

Let's keep as str for now.